sources/osgViewer/Viewer.js

```diff
@@ -193,8 +193,15 @@ Viewer.prototype = MACROUTILS.objectInherit( View.prototype, {
         return gl;
     },
 
+    setContextLostCallback: function ( cb ) {
+        this._contextLostCallback = cb;
+    },
     contextLost: function () {
         Notify.log( 'webgl context lost' );
+        if ( this._contextLostCallback ) {
+            this._contextLostCallback();
+        }
+        this._contextLost = true;
```

Comment:
It's better to initialize vars in the constructor, it's easier to know member variables used in the class. The few exceptions is when we need to call a init method that duplicate a lot of code initialized.